carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -36,1360 +36,1488 @@
-          traffic_light_states_[curr_intersection.id.id][current_movement_state.signal_group].back().first < min_end_time)
-      {
-        ROS_DEBUG_STREAM("Updated time for state: " << static_cast<lanelet::CarmaTrafficSignalState>(current_movement_state.movement_event_list[0].event_state.movement_phase_state)  << ", with time: " 
-                          << std::to_string(lanelet::time::toSec(min_end_time)));
-        traffic_light_states_[curr_intersection.id.id][current_movement_state.signal_group].back().first = min_end_time;
-        continue;
-      }
+        if (curr_intersection.moy_exists) //account for minute of the year
+        {
+          auto inception_boost(boost::posix_time::time_from_string("1970-01-01 00:00:00.000")); // inception of epoch
+          auto duration_since_inception(lanelet::time::durationFromSec(ros::Time::now().toSec()));
+          auto curr_time_boost = inception_boost + duration_since_inception;
+          ROS_DEBUG_STREAM("Calculated current time: " << boost::posix_time::to_simple_string(curr_time_boost));
+
+          int curr_year = curr_time_boost.date().year();
+          auto curr_year_start_boost(boost::posix_time::time_from_string(std::to_string(curr_year)+ "-01-01 00:00:00.000"));
+
+          ROS_DEBUG_STREAM("MOY extracted: " << (int)curr_intersection.moy);
+          auto curr_minute_stamp_boost = curr_year_start_boost + boost::posix_time::minutes((int)curr_intersection.moy);
+
+          int hours_of_day = curr_minute_stamp_boost.time_of_day().hours();
+          int curr_month = curr_minute_stamp_boost.date().month();
+          int curr_day = curr_minute_stamp_boost.date().day();
 
-      // detected that new state received; therefore, set the last recorded state (not new one received)
-      ROS_DEBUG_STREAM("Received new state for light: " << curr_light_id << ", with state: " << static_cast<lanelet::CarmaTrafficSignalState>(current_movement_state.movement_event_list[0].event_state.movement_phase_state) <<
-                        ", time: " << ros::Time::fromBoost(min_end_time));
+          auto curr_day_boost(boost::posix_time::time_from_string(std::to_string(curr_year) + "/" + std::to_string(curr_month) + "/" + std::to_string(curr_day) +" 00:00:00.000")); // GMT is the standard
+          auto curr_hour_boost = curr_day_boost + boost::posix_time::hours(hours_of_day);
 
+          min_end_time += lanelet::time::durationFromSec(lanelet::time::toSec(curr_hour_boost));
+          ROS_DEBUG_STREAM("New min_end_time: " << std::to_string(lanelet::time::toSec(min_end_time)));
+        }
+
+        auto last_time_difference = sim_.last_seen_state_[curr_intersection.id.id][current_movement_state.signal_group].first - min_end_time;  
+        bool is_duplicate = last_time_difference.total_milliseconds() >= -100 && last_time_difference.total_milliseconds() <= 100;
```

Comment:
Can you provide some reasoning as to why >= -100 and <= 100 is the check for duplicate time?

Reply:
There seem to have slight drift or change in the timers sent, so sometimes exact `==` doesn't work. So I thought it is safer to check if new value is within 0.1 seconds boundary

Reply:
Since 100ms is the period of spat messages in the ideal case, maybe a lower value would be better. Like 30 to avoid potentially ignoring a message as a false duplicate?

Reply:
reduced it to 30ms